newrelic/hooks/component_sklearn.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2010 New Relic, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from newrelic.api.function_trace import FunctionTrace
+from newrelic.api.transaction import current_transaction
+from newrelic.common.object_wrapper import wrap_function_wrapper
+
+
+def wrap_method(wrapped, instance, args, kwargs):
+    # If there is no transaction, do not wrap anything.
+    if not current_transaction():
+        return wrapped(*args, **kwargs)
+
+    method_name = wrapped.__name__
```

Comment:
`newrelic.common.object_names` includes `callable_name` and `object_context` functions that have more robust logic than a simple `__name__` lookup which may be missing or incorrect depending on what weird things a customer might choose to do.

If you want a fully qualified class path without the module included, you'll likely want `object_context(wrapped)[1]`. You can look at the comments on those functions for more info.

Reply:
I think this is a non-issue now though good to know for the future!